src/coreclr/jit/llvm.cpp

```diff
@@ -1181,9 +1205,27 @@ void Llvm::buildCast(GenTreeCast* cast)
         Value* intValue = _builder.CreateZExt(getGenTreeValue(cast->CastOp()), getLlvmTypeForVarType(TYP_INT));
         mapGenTreeToValue(cast, intValue); // nothing to do except map the source value to the destination GenTree
     }
-    else if (castToType == TYP_DOUBLE && cast->CastOp()->TypeIs(TYP_FLOAT))
+    else if (castToType == TYP_DOUBLE)
```

Comment:
This method is not easy to follow...

I think it should be restructured:
```c
var_types castFromType  = genActualType(cast->CastOp());
var_types castToType    = cast->CastToType();
Value*    castFromValue = consumeValue(castFromType, getLlvmTypeForVarType(castFromType));
Value*    castValue     = nullptr;

switch (castFromType)
{
    case TYP_INT:
    case TYP_LONG:
        switch (castToType) { ... }
        break;

    case TYP_FLOAT:
    case TYP_DOUBLE:
        switch (castToType) { ... }
        break;

    default:
        failFunctionCompilation(); // NYI
}

mapGenTreeToValue(castValue);
```

Reply:
Thanks